addons/account/models/account_accrued_entry.py

```diff
@@ -0,0 +1,78 @@
+import json
+from markupsafe import escape
+from odoo import models, _
+
+
+class AccruedEntryMixin(models.AbstractModel):
+    _name = 'account.accrued.entry.mixin'
+    _description = 'Account Accrued Entry Mixin'
+
+    def _get_preview_data(self, move_vals, currency_id=None):
+        preview_columns = [
+            {'field': 'account_id', 'label': _('Account')},
+            {'field': 'name', 'label': _('Label')},
+            {'field': 'debit', 'label': _('Debit'), 'class': 'text-end text-nowrap'},
+            {'field': 'credit', 'label': _('Credit'), 'class': 'text-end text-nowrap'},
+        ]
+        preview_vals = [self.env['account.move']._move_dict_to_preview_vals(move_vals, currency_id)]
+        return json.dumps({
+            'groups_vals': preview_vals,
+            'options': {
+                'columns': preview_columns,
+            },
+        })
+
+    def _get_aml_vals(self, label, balance, account_id, currency_id=None, amount_currency=None, analytic_distribution=None):
+        values = {
+            'name': label,
+            'debit': balance if balance > 0 else 0,
+            'credit': -balance if balance < 0 else 0,
+            'account_id': account_id,
+        }
+        if currency_id and amount_currency is not None:
+            values.update({
+                'currency_id': currency_id,
+                'amount_currency': amount_currency,
+            })
+        if analytic_distribution:
+            values.update({
+                'analytic_distribution': analytic_distribution,
+            })
+        return values
+
+    def _get_move_vals(self):
+        """
+        To be overriden as every accrued entry has their own specific process
+        """
+        return {}
+
+    def create_and_reverse_move(self):
+        move_vals = self._get_move_vals()
+        move = self.env['account.move'].create(move_vals)
+        move.action_post()
+        reverse_move = move._reverse_moves(default_values_list=[{
+            'ref': _("Reversal of: %s", move.ref),
+        }])
+        reverse_move.action_post()
+        return move, reverse_move
+
+    def create_wizard_view(self, view_vals):
+        view = {
+            'type': 'ir.actions.act_window',
+            'res_model': 'account.move',
+            'view_mode': 'tree,form',
+        }
+        if view_vals:
+            view.update(view_vals)
+        return view
```

Comment:
Would maybe make more sense to define this on `account.move` so that it can also switch easily between list view or only form view depending on the length of `self`